beamer/agent/chain.py

```diff
@@ -611,73 +615,76 @@ def maybe_prove(claim: Claim, context: Context) -> None:
     if not can_prove:
         return
 
-    if request.fill_tx is not None:
-        proof_tx = request.fill_tx
-    elif claim.invalidation_tx is not None:
-        proof_tx = claim.invalidation_tx
-
-    if proof_tx in context.l1_resolutions:
+    if proved_tx in context.l1_resolutions:
         return
 
+    assert proved_tx is not None
```

Comment:
We can remove this `assert` and move `if proved_tx in ...` check to just below line 609 so that we don't even do other checks if `proved_tx is None` or we already have a resolution for it:

```python
    if proved_tx is None or proved_tx in context.l1_resolutions:
        return
```

Reply:
sure